src/TuxState.js

```diff
@@ -0,0 +1,650 @@
+      //throw error if newPropsAtKey is not a number
+      invariantNumberCheck(newPropsAtKey);
+      //perform division on the corresponding deep keys
+      newState[key] = currentStateAtKey / newPropsAtKey;
+    });
+    //set state with the updated values and a callback (if provided), triggering a re-render
+    this.setState(newState, callback);
+  },
+
+  //omitState FUNCTION: removes the deepest keys in the passed in object from the corresponding deepest keys in the current state, or throws an error if keys don't match
+  //@param propsToOmit OBJECT: required object argument with deepest keys being booleans
+  //@param callback FUNCTION: optional callback that will be executed once replaceState is completed and the component is re-rendered
+  omitState: function (propsToOmit, callback) {
+    //throw error if propsToOmit is undefined or not an object
+    invariantArgCheck(propsToOmit);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    //build new state object
+    var newState = buildNewState(currentState, propsToOmit, function(newState, key, currentStateAtKey, newPropsAtKey) {
+      //remove the deepest key
+      delete newState[key];
+    });
+    //replace current state with the updated values and a callback (if provided), triggering a re-render
+    this.replaceState(newState, callback);
+  },
+
+  //extendState FUNCTION: adds new keys from the passed in object to the current state and overrides pre-existing keys, throws error if no outer keys in passed in object match outer keys in current state
+  //@param propsToExtend OBJECT: required object argument with at least one outer key matching an outer key in the current state, and any additional keys being keys to add to the current state
+  //@param callback FUNCTION: optional callback that will be executed once setState is completed and the component is re-rendered
+  extendState: function (propsToExtend, callback) {
+    //throw error if propsToExtend is undefined or not an object
+    invariantArgCheck(propsToExtend);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    var matchedKey = false;
+    //check that at least one outer key in the passed in object matches an outer key in the current state
+    for (var key in propsToExtend) {
+      if (currentState[key]) {
+        //if we find any key that maches a key in the current state, set matchedKey to true
+        matchedKey = true;
+      }
+    }
+    //throw error if no keys match any keys in the current state
+    invariant(matchedKey, "At least one outer key must match an outer key in the current state. Use setState if you only wish to add new keys and not change existing keys.");
+    //build new object with all keys in the current state plus any additional keys in propsToExtend, overwriting any matching keys with those in propsToExtend
+    var newState = assign({}, currentState, propsToExtend);
```

Comment:
We still need to use our deep recursion for extendState so that we dont objects within the keys at these objects.